File: Graph/Python/2331 반복수열.py
a, p = map(int, input().split())
arr = []

def calculator(num) :
    result = 0
    while num != 0 :
        
        n = num%10
        n = int(pow(n, p))
        result += n
        num = num//10
    return result
check = -1
arr.append(a)
value = calculator(arr[-1])
while True :
    if value in arr and check == -1 :
        check = value
        break
    else :
        arr.append(value)
        value = calculator(arr[-1])

index = arr.index(check)
print(len(arr[:index]))

File: Graph/Kotlin/2331 반복수열.kt
import java.util.Scanner
import kotlin.math.*


var p: Int = 0

fun cal(num : Int):Int {
    
    var n : Int = num
    var result = 0;
    while(n != 0) {
        var temp = 0
        temp = (n%10.toFloat()).pow(p).toInt()
        result += temp
        n = n/10
    }
    return result
}


fun main(args: Array<String>) {
    var sc = Scanner(System.`in`)
    var a = sc.nextInt()
    p = sc.nextInt()
    var check = -1
    
    var arr = arrayListOf<Int>(a)
    
    while(true) {
        var value = cal(arr[arr.size-1])
        
        if(arr.contains(value)) {
            check = value
            break
        }
        arr.add(value)
    }
    
    println(arr.indexOf(check))
}
